### PR TITLE
Add question about how kernel modules are signed

### PR DESCRIPTION
**Pull request by julian-klode**
At least Debian is currently signing modules with the same key as the kernel, causing the kernel to be able to load modules not intended for it, potentially causing grave security issues. This is violating the security model, so let's make sure nobody does stuff like that anymore.

### CODE REVIEW

**Review by jsetje — APPROVED** [review]

File: README.md
Comment (xnox) [inline README.md:151]: I disagree with this requirement.

Ubuntu does not only use ephemeral key for signing kernel modules. And it by default trusts: ephemeral build-time, 3rd-party driver signing key, Livepatch signing key, and MOK signing keys.

In practice MOK signing key is always the same for all kernels.

Which key is used and trusted, and whether it is chained to CA is not at all interesting.

Instead the following rules should be followed instead
1. Ensure unique modversions are set for each distinct kernel build https://www.kernel.org/doc/html/latest/admin-guide/module-signing.html#administering-protecting-the-private-key
2. Ensure that lockdown=integrity by default prohibits bypassing --force-vermagic

This is easily checked by comparing two distinct abi builds, running modinfo on the modules from both and checking that vermagic is different.

Note that there are additional abi checks that prevent modules loading from incompatible builds of kernel on top of that (but those use non-cryptographically secure CRC).

I request to revert this requirement, as it is crafted without full consideration of existing keys in use for existing secureboot signed keys in all kernels from all vendors, which today all use the same key for different abis and already prevent ability to load a kernel module from one kernel abi on the other abi. If that is incorrect for any vendor, it must be resolved by correctly bumping EXTRAVERSION/LOCALVERSION in every published build.

Also we should ensure all certificates used for kernel module signing, have the extension set that such certs are for module signing only.

**Review by xnox — COMMENTED** [review]
I disagree with premise and wording of this requirement, as it does not at all address correctly the concern raised.

File: README.md
Comment (xnox) [reply README.md:151]: cc: @jsetje @steve-mcintyre i disagree with this wording being merged, without reviews or approvals from at least 2 different kernel maintainers. and the proposer of this change is not a kernel maintainer themselves.

Comment (julian-klode) [reply README.md:151]: I believe modversions is the wrong thing, because `CONFIG_MODVERSIONS` is the CRC mechanism, you want to ensure each kernel build has a separate `EXTRAVERSION` or `CONFIG_LOCALVERSION`.

I'm wary of allowing people to reuse kernel ABIs because they *think* it is still the same, they can't really proof it or have the tooling, hence generally guiding them to something that works is what I'm looking for.

Comment (dennis-tseng99) [reply README.md:151]: I'm curious about the memory life or size of server. Instead of using Canonical key, for each Livepatch operation, MOK needs to add into or erase from the database of keys, which might reduce the usage life of NVRAM.

**Pull request merged**
